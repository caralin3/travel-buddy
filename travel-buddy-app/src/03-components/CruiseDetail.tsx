import moment from 'moment';
import React from 'react';
import { Col, Row } from 'reactstrap';
import { Cruise } from '../api';
import { getDurationDays } from '../utils';

export interface CruiseDetailProps {
  cruise: Cruise;
  end?: boolean;
}

export const CruiseDetail: React.FC<CruiseDetailProps> = ({
  cruise: {
    cabinNumber,
    cruiseLine,
    cabinType,
    departureCity,
    departureCountry,
    departureState,
    destinationCity,
    destinationCountry,
    destinationState,
    endDate,
    roundTrip,
    shipName,
    startDate,
  },
  end = false,
}) => {
  const days = getDurationDays(startDate, endDate);

  return (
    <div className="cruise-detail">
      <Row>
        <Col className="cruise-detail__name text-primary">
          {cruiseLine}: {shipName}
        </Col>
      </Row>
      <Row className="my-2">
        <Col xs={12} md={4}>
          <p className="cruise-detail__label">{end ? 'Arrive' : 'Depart'}</p>
          {end && !roundTrip ? (
            <p className="m-0">
              {destinationCity}, {!!destinationState ? destinationState : destinationCountry} at{' '}
              {moment(endDate).format('h:mm A')}
            </p>
          ) : (
            <p className="m-0">
              {departureCity}, {!!departureState ? departureState : departureCountry} at{' '}
              {moment(end ? endDate : startDate).format('h:mm A')}
            </p>
          )}
        </Col>
        <Col xs={12} md={2}>
          <p className="cruise-detail__label">Duration</p>
          <p className="m-0">
            {days} day{days > 1 ? 's' : ''}
            {roundTrip ? ' (roundtrip)' : ''}
          </p>
        </Col>
        {!end && (
          <Col xs={12} md={6}>
            <p className="cruise-detail__label">Notes</p>
            <p className="cruise-detail__notes m-0">
              Cabin #{cabinNumber} {cabinType ? `(${cabinType.slice(0, 1)}${cabinType.slice(1).toLowerCase()})` : ''}
            </p>
          </Col>
        )}
      </Row>
    </div>
  );
};
