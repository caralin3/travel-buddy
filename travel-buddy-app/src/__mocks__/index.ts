export * from './activities';
export * from './cruises';
export * from './flights';
export * from './hotels';
export * from './trips';
export * from './user';
