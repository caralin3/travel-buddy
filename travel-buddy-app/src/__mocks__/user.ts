import { User } from '../api';

export const user: User = {
  id: 0,
  firstName: 'Jane',
  lastName: 'Doe',
  email: 'janeD@email.com',
  roles: [
    {
      id: 0,
      name: 'ROLE_USER',
    },
  ],
};
