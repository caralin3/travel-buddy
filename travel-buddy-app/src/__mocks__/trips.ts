import { Trip } from '../api';
import { user } from './user';

export const trips: Trip[] = [
  {
    user,
    id: 0,
    title: 'Texas Girls Trip 2022',
    description: 'Travelers: Cara, Elizabeth, Sarah<br/>Destination: Dallas, Texas',
    startDate: '2022-06-03',
    endDate: '2022-06-06',
    uniqueLink: '',
  },
  {
    user,
    id: 1,
    title: 'Trip 2',
    startDate: '2022-08-14',
    endDate: '2022-08-31',
  },
];
