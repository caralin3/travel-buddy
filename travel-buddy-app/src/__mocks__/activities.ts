import { Activity } from '../api';
import { trips } from './trips';

export const activities: Activity[] = [
  {
    id: 0,
    name: 'Horseback Riding',
    description: 'Horseback riding at Stockyard stables.',
    startDate: '2022-06-04 11:00:00',
    endDate: '2022-06-04 12:00:00',
    cost: 75,
    currency: 'USD',
    addressLine1: '157 E Exchange Ave',
    city: 'Fort Worth',
    state: 'Texas',
    country: 'USA',
    postalCode: '76164',
    trip: trips[0],
  },
  {
    id: 1,
    name: 'Billy Bobs Texas',
    description: 'Honky Tonk near Stockyard stables.',
    startDate: '2022-06-04 16:00:00',
    endDate: '2022-06-04 18:00:00',
    cost: 20,
    currency: 'USD',
    addressLine1: '2520 Rodeo Plaza',
    city: 'Fort Worth',
    state: 'Texas',
    country: 'USA',
    postalCode: '76164',
    trip: trips[0],
  },
  {
    id: 2,
    name: 'The Sixth Floor Museum at Dealey Plaza',
    description: 'JFK Museum',
    startDate: '2022-06-05 13:30:00',
    endDate: '2022-06-05 15:00:00',
    cost: 1,
    currency: 'USD',
    addressLine1: '411 Elm Street',
    city: 'Dallas',
    state: 'Texas',
    country: 'USA',
    postalCode: '75202',
    trip: trips[0],
  },
  {
    id: 3,
    name: "Adair's Saloon",
    description:
      'Same area=Dallas World Aquarium, 6th floor museum and dealy plaza, and Museum of Nature and Science SE of hotel ',
    startDate: '2022-06-03 16:00:00',
    endDate: '2022-06-03 18:00:00',
    cost: 20,
    currency: 'USD',
    addressLine1: '2624 Commerce Street',
    city: 'Dallas',
    state: 'Texas',
    country: 'USA',
    postalCode: '75226',
    trip: trips[0],
  },
];
