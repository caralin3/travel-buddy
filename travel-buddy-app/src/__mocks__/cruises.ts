import { Cruise, Port } from '../api';
import { trips } from './trips';

export const cruises: Cruise[] = [
  {
    id: 0,
    cabinType: 'SUITE',
    cruiseLine: 'Princess Cruises',
    shipName: 'Discovery Princess',
    cost: 3000,
    currency: 'USD',
    cabinNumber: '11234',
    startDate: '2022-05-08 16:00:00',
    endDate: '2022-05-15 07:00:00',
    description: 'Alaskan Cruise',
    departureCity: 'Seattle',
    departureState: 'Washington',
    departureCountry: 'US',
    roundTrip: true,
    trip: trips[0],
  },
  {
    id: 1,
    cabinType: 'DOUBLE',
    cruiseLine: 'Royal Caribbean',
    shipName: 'Discovery Princess',
    cost: 5402.97,
    currency: 'USD',
    cabinNumber: '9112',
    startDate: '2025-04-12',
    endDate: '2025-04-19',
    departureCity: 'Sicily',
    departureCountry: 'Italy',
    destinationCity: 'Athens',
    destinationCountry: 'Greece',
    roundTrip: false,
    trip: trips[0],
  },
];

export const cruise1Ports: Port[] = [
  {
    id: 0,
    day: 2,
    arrival: '2022-05-10 06:30:00',
    departure: '2022-05-10 15:00:00',
    city: 'Ketchikan',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 1,
    day: 3,
    arrival: '2022-05-11 12:30:00',
    departure: '2022-05-11 21:00:00',
    city: 'Juneau',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 2,
    day: 4,
    arrival: '2022-05-12 06:00:00',
    departure: '2022-05-12 17:00:00',
    city: 'Skagway',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 3,
    day: 6,
    arrival: '2022-05-14 19:00:00',
    departure: '2022-05-14 23:59:00',
    city: 'Victoria',
    country: 'Canada',
    cruise: cruises[0],
  },
];

export const cruise2Ports: Port[] = [
  {
    id: 0,
    day: 2,
    arrival: '2025-04-14 06:30:00',
    departure: '2025-04-14 15:00:00',
    city: 'Florence',
    country: 'Italy',
    cruise: cruises[1],
  },
  {
    id: 1,
    day: 5,
    arrival: '2025-04-17 12:30:00',
    departure: '2025-04-17 21:00:00',
    city: 'Rome',
    country: 'Italy',
    cruise: cruises[1],
  },
  {
    id: 2,
    day: 6,
    arrival: '2025-04-18 06:00:00',
    departure: '2025-04-18 17:00:00',
    city: 'Santorini',
    country: 'Greece',
    cruise: cruises[1],
  },
];
