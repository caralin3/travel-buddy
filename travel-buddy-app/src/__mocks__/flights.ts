import { Flight } from '../api';
import { trips } from './trips';

export const flights: Flight[] = [
  {
    id: 1,
    airline: 'JetBlue',
    arrivalAirport: 'JFK',
    arrivalCity: 'NYC',
    arrivalDate: '2022-06-03 15:30:00',
    confirmationCode: 'FWD5DBJ',
    departureAirport: 'DFW',
    departureCity: 'Dallas',
    departureDate: '2022-06-06 13:05:00',
    flightClass: 'ECONOMY',
    flightNumber: '124',
    seats: '12 DEF',
    terminal: 'E',
    trip: trips[0],
  },
  {
    id: 0,
    airline: 'JetBlue',
    arrivalAirport: 'DFW',
    arrivalCity: 'Dallas',
    arrivalDate: '2022-06-03 13:39:00',
    confirmationCode: 'FWD5DBJ',
    cost: 300.03,
    currency: 'USD',
    departureAirport: 'JFK',
    departureCity: 'NYC',
    departureDate: '2022-06-03 10:05:00',
    flightClass: 'ECONOMY',
    flightNumber: '123',
    terminal: '5',
    seats: '7 ABC',
    trip: trips[0],
  },
];
