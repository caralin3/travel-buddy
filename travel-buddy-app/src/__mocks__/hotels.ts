import { Hotel } from '../api';
import { trips } from './trips';

export const hotels: Hotel[] = [
  {
    id: 0,
    name: 'Courtyard By Marriot Denton',
    roomType: 'DOUBLE',
    roomCount: 1,
    checkInDate: '2022-06-03 15:00:00',
    checkOutDate: '2022-06-06 11:00:00',
    cost: 0,
    currency: 'USD',
    addressLine1: '2800 Colorado Blvd',
    city: 'Denton',
    state: 'TX',
    trip: trips[0],
  },
];
