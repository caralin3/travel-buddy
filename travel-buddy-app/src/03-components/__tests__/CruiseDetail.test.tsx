import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { cruises } from '../../__mocks__';
import { CruiseDetail } from '../CruiseDetail';

describe('CruiseDetail', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<CruiseDetail cruise={cruises[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render with end correctly', () => {
    const wrapper = shallow(<CruiseDetail end cruise={cruises[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render with roundtrip correctly', () => {
    const wrapper = shallow(
      <CruiseDetail cruise={{ ...cruises[1], endDate: '2022-03-06', startDate: '2022-03-06' }} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render with end and without roundtrip correctly', () => {
    const wrapper = shallow(<CruiseDetail end cruise={cruises[1]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
