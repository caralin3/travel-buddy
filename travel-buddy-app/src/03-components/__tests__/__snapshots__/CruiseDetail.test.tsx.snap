// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CruiseDetail should render correctly 1`] = `
<div
  className="cruise-detail"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      className="cruise-detail__name text-primary"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      Princess Cruises
      : 
      Discovery Princess
    </Col>
  </Row>
  <Row
    className="my-2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Depart
      </p>
      <p
        className="m-0"
      >
        Seattle
        , 
        Washington
         at
         
        4:00 PM
      </p>
    </Col>
    <Col
      md={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Duration
      </p>
      <p
        className="m-0"
      >
        7
         day
        s
         (roundtrip)
      </p>
    </Col>
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Notes
      </p>
      <p
        className="cruise-detail__notes m-0"
      >
        Cabin #
        11234
         
        (Suite)
      </p>
    </Col>
  </Row>
</div>
`;

exports[`CruiseDetail should render with end and without roundtrip correctly 1`] = `
<div
  className="cruise-detail"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      className="cruise-detail__name text-primary"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      Royal Caribbean
      : 
      Discovery Princess
    </Col>
  </Row>
  <Row
    className="my-2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Arrive
      </p>
      <p
        className="m-0"
      >
        Athens
        , 
        Greece
         at
         
        12:00 AM
      </p>
    </Col>
    <Col
      md={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Duration
      </p>
      <p
        className="m-0"
      >
        1
         day
      </p>
    </Col>
  </Row>
</div>
`;

exports[`CruiseDetail should render with end correctly 1`] = `
<div
  className="cruise-detail"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      className="cruise-detail__name text-primary"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      Princess Cruises
      : 
      Discovery Princess
    </Col>
  </Row>
  <Row
    className="my-2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Arrive
      </p>
      <p
        className="m-0"
      >
        Seattle
        , 
        Washington
         at
         
        7:00 AM
      </p>
    </Col>
    <Col
      md={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Duration
      </p>
      <p
        className="m-0"
      >
        7
         day
        s
         (roundtrip)
      </p>
    </Col>
  </Row>
</div>
`;

exports[`CruiseDetail should render with roundtrip correctly 1`] = `
<div
  className="cruise-detail"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      className="cruise-detail__name text-primary"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      Royal Caribbean
      : 
      Discovery Princess
    </Col>
  </Row>
  <Row
    className="my-2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Depart
      </p>
      <p
        className="m-0"
      >
        Sicily
        , 
        Italy
         at
         
        12:00 AM
      </p>
    </Col>
    <Col
      md={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Duration
      </p>
      <p
        className="m-0"
      >
        1
         day
      </p>
    </Col>
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="cruise-detail__label"
      >
        Notes
      </p>
      <p
        className="cruise-detail__notes m-0"
      >
        Cabin #
        9112
         
      </p>
    </Col>
  </Row>
</div>
`;
